### DISCUSSION TITLE
Pinch and zoom on mobile

### DISCUSSION THREAD

**Discussion by tonyneel923** [airbnb/visx#1292] [Q&A] [answered]
I have built a map using Zoom and Mercator which works well on desktop but does not support mobile pinch and zoom. I also do not see a way to do it in visx/zoom.

Am I missing something?

Thanks for any help!

**Comment by williaster** [comment]
Hey @tonyneel923 👋   this is a great question, haven't seen it pop up before. Currently `Zoom` only supports [wheel events to change scale](https://github.com/airbnb/visx/blob/master/packages/visx-zoom/src/Zoom.tsx#L240). I think this would need to be explicitly added as a feature, it might be worth considering using something like [`react-use-gesture`](https://github.com/pmndrs/use-gesture) to abstract some of the complexity 🤔 .

We could convert this to an issue to track it as a feature. I don't have the bandwidth to do it at the moment but definitely support it and happy to review/get it in.

**Reply by tonyneel923** [reply]
Thanks for the reply @williaster. I will try out react-use-gesture and post a working example here if I get it to work. I will take a peek at the code after and see if I can get a pr up for it hopefully. Would a pr with the above library as a dep be ok?

**Reply by tonyneel923** [reply]
I just glanced at the code for Zoom.tsx and it seems to be a class component and not compatible with hooks. So any work there would need to convert it if we use react-use-gesture.

**Comment by tonyneel923** [comment, marked as answer]
@williaster I have submitted a PR here https://github.com/airbnb/visx/pull/1305
